File: kumoh-code-2017/Java/#1 기본/#2 수업/소스코드/11. 은행관리 프로그램 (완성본)/은행관리/MinusAccount.java
package 은행관리;

public class MinusAccount extends Account
{
	private long money;	
	
	public MinusAccount(int account_number, int pin, long amount)
	{
		super(account_number, pin, amount);
	}
	
	public void credit(long amount)
	{
		money = this.getBalance() + amount;
		if(money > 0)
		{
			money = (long)(money * 1.01);
			plusBalance(money);
		}
		else
		{
			money = (long)(money * 1.05);
			plusBalance(money);
		}
	}
	
	public void debit(long amount)
	{
		money = this.getBalance() - amount;
		if(money > 0)
		{
			money = (long)(money * 1.01);
			minusBalance(money);
		}
		else
		{
			money = (long)(money * 1.05);
			minusBalance(money);
		}
	}
}

File: kumoh-code-2017/C++/#2 설계/과제15 은행관리 프로그램/MinusAccount.cpp
#include"MinusAccount.h"

MinusAccount::MinusAccount(int accountNumber, int password, int balance) :Account(accountNumber, password, balance){

}

//====메소드 재정의 (다형성) =====

void MinusAccount::credit(int money){
	balance = balance + money;
}

void MinusAccount::debit(int money){
	balance = balance - money;
}
